### PR TITLE
Switch sits import from XML to API.kent  (powered by mssql view)

### PR DESCRIPTION
**Pull request by jna-unikent**
This merge requests canges the sitsimport script so that consumes deleievery (or programmeheader) data from the kent API.

This relies on: https://gitlab.kent.ac.uk/webdev/api.kent/merge_requests/278

### CODE REVIEW

File: application/tasks/sitsimport.php [hunk 1]
```diff
@@ -126,20 +126,6 @@ public function getCurrentYear( $level ) {
     return Setting::get_setting( $level . "_current_year" );
   }
 
-  public function checkCourseIsValid( $course ) {
```
Comment (sy238-unikent) [inline application/tasks/sitsimport.php]: This is now filtered out in the api before this data ever gets here.

File: application/tasks/sitsimport.php [hunk 2]
```diff
@@ -126,20 +126,6 @@ public function getCurrentYear( $level ) {
...
-    if ( $course->progID == '' ) {
-      return false;
-    }
-    return true;
-  }
-
-  public function checkIPOIsValid( $ipo ) {
```
Comment (sy238-unikent) [inline application/tasks/sitsimport.php]: This is now filtered out in the api before this data ever gets here.

File: application/tasks/sitsimport.php [hunk 3]
```diff
@@ -31,50 +31,43 @@ public function run( $args = array() ) {
...
-    // Load XML file
-    $xml = $this->loadXML();
-
-    // If XML file is good, purge old caches before starting import
-    $this->purgeOldPGData($this->processYears['pg']);
-    $this->purgeOldUGData($this->processYears['ug']);
-
-    foreach ( $xml as $course ) {
-      //make sure it has a programme ID in the progsplant
-      if ( !$this->checkCourseIsValid( $course ) ) {
```
Comment (sy238-unikent) [inline application/tasks/sitsimport.php]: no longer needed, check is done at api db query level

File: application/tasks/sitsimport.php [hunk 4]
```diff
@@ -1,170 +1,213 @@
...
+ * This task retrieves data about programmes from SITS via api.kent
+ * and imports the data into the Programmes Plant (pg|ug)_programme_deliveries tables.
+ *
+ * A logfile is generated (purged at the start of each request) in {storage}/logs/sits_import.log.
+ *
+ * Another script "sitsimport_check.php" can be run to check if this log file contains any errors.
  *
- * Data imported is IPO, POS code, MCR code, ARI code, description,
- * award and attendance type (full-time or part-time)
+ * The data imported is:
```
Comment (cfc7-unikent) [inline application/tasks/sitsimport.php]: this is nice info

**Pull request merged**